src/js/shield.js

```diff
@@ -137,72 +145,51 @@ function textColor(shieldDef) {
 
 function getDrawFunc(shieldDef) {
   if (typeof shieldDef.canvasDrawnBlank != "undefined") {
-    return (ref) =>
+    return (ctx, ref) =>
       ShieldDraw.draw(
         shieldDef.canvasDrawnBlank.drawFunc,
+        ctx,
         shieldDef.canvasDrawnBlank.params,
         ref
       );
   }
   return ShieldDraw.blank;
 }
 
-/**
- * Creates a graphics context of the correct size to hold the shield and banner.
- * @param {*} shieldDef shield definition
- * @param {*} routeDef route definition
- * @returns a blank graphics context
- */
-function generateBlankGraphicsContext(sprites, shieldDef, routeDef) {
-  var bannerCount = getBannerCount(shieldDef);
-  var shieldArtwork = getRasterShieldBlank(sprites, shieldDef, routeDef);
-  var compoundBounds = null;
-
-  if (shieldArtwork == null) {
-    let drawFunc = getDrawFunc(shieldDef);
-    let drawnShieldCtx = drawFunc(routeDef.ref);
-    compoundBounds = compoundShieldSize(drawnShieldCtx.canvas, bannerCount);
-  } else {
-    compoundBounds = compoundShieldSize(shieldArtwork.data, bannerCount);
-  }
+function drawShield(ctx, shieldDef, routeDef) {
+  let bannerCount = getBannerCount(shieldDef);
+  let yOffset = bannerCount * ShieldDef.bannerSizeH + ShieldDef.topPadding;
 
-  return Gfx.getGfxContext(compoundBounds);
+  //Shift canvas to draw shield below banner
+  ctx.translate(0, yOffset);
+  let drawFunc = getDrawFunc(shieldDef);
+  drawFunc(ctx, routeDef.ref);
+  ctx.translate(0, -yOffset);
```

Comment:
Better to bracket these with [save]…restore:

[save]: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/save

```suggestion
  ctx.save();
  ctx.translate(0, yOffset);
  let drawFunc = getDrawFunc(shieldDef);
  drawFunc(ctx, routeDef.ref);
  ctx.restore();
```